lib/chat_app/structs/message.ex

```diff
@@ -0,0 +1,35 @@
+defmodule ChatApp.Structs.Message do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  @primary_key {:message_id, Ecto.UUID, autogenerate: true}
+  @foreign_key_type Ecto.UUID
+
+  schema "messages" do
+    field(:sender, :string)
+    field(:content, :string)
+    belongs_to :room, ChatApp.Structs.Room,
+      foreign_key: :room_id,
+      references: :room_id
```

Comment:
do you need to specify the foreign key explicitly?
Afaik the FK name is inferred by the singular name of the referenced schema + `_id`.

Reply:
I tried without specifying and it told me ":id" key does not exist, which is weird, and this workaround is what I found in the Northwind Etso Model

Reply:
Maybe I can try without specifying again and see the problem together?

Reply:
So should I modify the PKs to be id instead? I did them this way to avoid ambiguity in case we have multiple entities that all have "id". Although I guess it's hard to mismatch since most of the calls are with the struct specified as well

Reply:
@Davis856  yeah, modify the PKs to be named `id`.